src/main/kotlin/com/github/minecraftschurlimods/helperplugin/HelperExtension.kt

```diff
@@ -240,39 +231,54 @@ open class HelperExtension @Inject constructor(private val project: Project) {
         dataGenSourceSet = project.sourceSets.maybeCreate("data")
         project.dependencies {
             dataGenSourceSet.implementationConfigurationName(project.sourceSets.main.output)
-            dataGenSourceSet.implementationConfigurationName(neoforge())
+            project.neoForge.addModdingDependenciesTo(dataGenSourceSet)
+            // TODO is the line above a suitable replacement? dataGenSourceSet.implementationConfigurationName(neoforge())
             if (::apiSourceSet.isInitialized) {
                 dataGenSourceSet.implementationConfigurationName(apiSourceSet.output)
             }
         }
-        project.runs.matching { it.name == "data" }.all {
-            modSources.add(dataGenSourceSet)
-            idea.primarySourceSet = dataGenSourceSet
+        project.neoForge.mods.matching { it.name == "for_data_run" }.all {
+            sourceSet(dataGenSourceSet)
+        }
+        project.neoForge.runs.matching { it.name == "data" }.all {
+            sourceSet = dataGenSourceSet
         }
     }
 
     fun withCommonRuns() {
-        project.runs {
+        val dataModModel = project.neoForge.mods.create("for_data_run")
+        dataModModel.sourceSet(project.sourceSets.main.get())
+        val otherRunsModModel = project.neoForge.mods.create("for_other_runs")
+        otherRunsModModel.sourceSet(project.sourceSets.main.get())
+
+        project.neoForge.runs {
             all {
-                workingDirectory.set(project.layout.projectDirectory.dir("run"))
+                gameDirectory.set(project.layout.projectDirectory.dir("run"))
                 systemProperties.put("forge.logging.markers", "REGISTRIES")
                 systemProperties.put("forge.logging.console.level", "debug")
                 if (!runningInCI.getOrElse(false)) {
                     jvmArgument("-XX:+AllowEnhancedClassRedefinition")
                 }
-                modSources.add(project.sourceSets.main.get())
+                if (name == "data") {
+                    mods = listOf(dataModModel)
+                } else {
+                    mods = listOf(otherRunsModModel)
+                }
+            }
+            create("client") {
+                client()
             }
-            maybeCreate("client")
-            maybeCreate("server").apply {
-                singleInstance()
+            create("server") {
+                server()
+                // TODO singleInstance()
```

Comment:
Not (yet) supported by MDG as far as I know. Does this only control the IDEA run? If so it should be trivial to add.